src/SFA.DAS.Campaign.Web.UnitTests/ViewComponents/WhenRenderingTheButtonViewComponent.cs

```diff
@@ -0,0 +1,271 @@
+            var buttonType = ButtonType.Input;
+            //Act
+            var actual = await _sut.InvokeAsync(buttonType, ButtonStyle.Apprentice, _buttonName, buttonText, shadow: false, sparks: false);
+
+            //Assert
+            Assert.IsNotNull(actual);
+            var result = actual as ViewViewComponentResult;
+            Assert.IsNotNull(result);
+
+            Assert.AreEqual("Input", result.ViewName);
+
+            Assert.IsInstanceOf<ButtonViewModel>(result.ViewData.Model);
+            var buttonViewModel = (ButtonViewModel)result.ViewData.Model;
+            Assert.AreEqual(buttonType, buttonViewModel.Type);
+        }
+
+        [Test]
+        public async Task When_Button_Type_Is_Button_Then_The_Correct_View()
+        {
+            var buttonText = "Primary";
+            var buttonType = ButtonType.Button;
+            //Act
+            var actual = await _sut.InvokeAsync(buttonType, ButtonStyle.Apprentice, _buttonName, buttonText, shadow: false, sparks: false);
+
+            //Assert
+            Assert.IsNotNull(actual);
+            var result = actual as ViewViewComponentResult;
+            Assert.IsNotNull(result);
+
+            Assert.AreEqual("Button", result.ViewName);
+
+            Assert.IsInstanceOf<ButtonViewModel>(result.ViewData.Model);
+            var buttonViewModel = (ButtonViewModel)result.ViewData.Model;
+            Assert.AreEqual(buttonType, buttonViewModel.Type);
+        }
+
+        [Test]
+        public async Task When_Href_Supplied_Then_Model_Href_Populated()
+        {
+            var buttonText = "Primary";
+            var buttonType = ButtonType.Button;
+            string href = "/index.html";
+            //Act
+            var actual = await _sut.InvokeAsync(buttonType, ButtonStyle.Apprentice, _buttonName, buttonText, href: href );
+
+            //Assert
+            Assert.IsNotNull(actual);
+            var result = actual as ViewViewComponentResult;
+            Assert.IsNotNull(result);
+
+            Assert.IsInstanceOf<ButtonViewModel>(result.ViewData.Model);
+            var buttonViewModel = (ButtonViewModel)result.ViewData.Model;
+            Assert.AreEqual(href, buttonViewModel.Href);
+        }
+
+        [Test]
+        public async Task When_No_Href_Supplied_Then_Model_Href_Default()
+        {
+            var buttonText = "Primary";
+            var buttonType = ButtonType.Button;
+            string href = "#";
+            //Act
+            var actual = await _sut.InvokeAsync(buttonType, ButtonStyle.Apprentice, _buttonName, buttonText);
+
+            //Assert
+            Assert.IsNotNull(actual);
+            var result = actual as ViewViewComponentResult;
+            Assert.IsNotNull(result);
+
+            Assert.IsInstanceOf<ButtonViewModel>(result.ViewData.Model);
+            var buttonViewModel = (ButtonViewModel)result.ViewData.Model;
+            Assert.AreEqual(href, buttonViewModel.Href);
+        }
+    }
+}
```

Comment:
I can imagine this test file becoming extremely flakey and a bit of an overhead to maintain. I'm not sure if we are preserving the separation of front and back end code.